File: app/src/main/java/com/akay/fitnass/di/AppModule.java
package com.akay.fitnass.di;

import android.content.Context;

import com.akay.fitnass.view.notification.NotificationController;
import com.akay.fitnass.view.notification.NotificationControllerImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
@Singleton
public class AppModule {
    private final Context mContext;

    public AppModule(Context context) {
        this.mContext = context;
    }

    @Provides @Singleton
    Context provideContext() {
        return mContext;
    }

    @Provides @Singleton
    public NotificationController provideNotificationController(Context context) {
        return new NotificationControllerImpl(context);
    }
}

File: app/src/main/java/com/akay/fitnass/di/AppModule.kt
package com.akay.fitnass.di

import android.content.Context

import com.akay.fitnass.view.notification.NotificationController
import com.akay.fitnass.view.notification.NotificationControllerImpl

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
@Singleton
class AppModule(private val mContext: Context) {

    @Provides
    @Singleton
    internal fun provideContext(): Context = mContext

    @Provides
    @Singleton
    fun provideNotificationController(context: Context): NotificationController = NotificationControllerImpl(context)
}
